srml/randomness/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Randomness Module
+//!
+//! The Randomness module provides a [`random`](./struct.Module.html#method.random) function that
+//! generates low-influence random values based on the block hashes from the previous 81 blocks.
+//! Low-influence randomness can be useful when defending against relatively weak adversaries.
+//!
+//! ## Public Functions
+//!
+//! See the [`Module`](./struct.Module.html) struct for details of publicly available functions.
+//!
+//! ## Usage
+//!
+//! ### Prerequisites
+//!
+//! Import the Randomness module and derive your module's configuration trait from the system trait.
+//!
+//! ### Example - Get random seed for the current block
+//!
+//! ```
+//! use support::{decl_module, dispatch::Result};
+//!
+//! pub trait Trait: system::Trait {}
+//!
+//! decl_module! {
+//! 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+//! 		pub fn random_module_example(origin) -> Result {
+//! 			let _random_seed = <srml_randomness::Module<T>>::random_seed();
+//! 			Ok(())
+//! 		}
+//! 	}
+//! }
+//! # fn main() { }
+//! ```
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::{prelude::*, convert::TryInto};
+use sr_primitives::traits::Hash;
+use support::{decl_module, decl_storage};
+use safe_mix::TripletMix;
+use codec::Encode;
+use system::Trait;
+
+
+fn block_number_to_index<T: Trait>(block_number: T::BlockNumber) -> usize {
+	// '- 2' because on_initialize is called on the first block after genesis - block 2
+	let index = (block_number - 2.into()) % 81.into();
```

Comment:
Hmm, I think that the genesis number is `0`.

Reply:
I was basing my tests off of: https://github.com/paritytech/substrate/blob/094cca7b7726ef407cd82e65efe4ab42fff33801/srml/finality-tracker/src/lib.rs#L350-L368

which made me assume that it started at 1. I'll change this.